lib/puppet/face/module/generate.rb

```diff
@@ -129,13 +129,17 @@
       puts _("Finished; module generated in %{path}.") % { path: path }
       result.join("\n")
     end
+
+    deprecate
   end
 end
 
 module Puppet::ModuleTool::Generate
   module_function
 
   def generate(metadata, skip_interview = false)
+    Puppet.deprecation_warning _("`puppet module generate` is deprecated and will be removed in a future release. This action has been replaced by Puppet Development Kit. For more information visit https://puppet.com/download-puppet-development-kit.")
```

Comment:
I'm in two minds of adding another link here to the [specific `pdk new module` docs](https://puppet.com/docs/pdk/1.0/pdk_generating_modules.html#creating-modules-and-classes-with-pdk).

Reply:
I think no, because if a user hits this, I assume they need to download PDK. The `new module` docs are only useful if they've already download PDK , in which case I think (and hope) that it's unlikely they are using `puppet module generate`.